pkg/common/cns-lib/volume/manager.go

```diff
@@ -416,12 +420,8 @@ func (m *defaultManager) MonitorCreateVolumeTask(ctx context.Context,
 			}
 			log.Errorf("volume with name %s not present in CNS. Marking task %s as failed.",
 				volNameFromInputSpec, task.Reference().Value)
-			// Set reserved to 0 when task has failed so that the reserved
```

Comment:
All the `UpdateReservedInQuotaDetails` callers have been removed and the `reserved` field in CNSVolumeOperationRequest CRs will be set to 0 in the defer functions for Create and Expand APIs whenever the corresponding CNS task status is set to failed.